device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -1216,4 +1221,101 @@ private void verifyMethodsAreSupported()
             throw new UnsupportedOperationException(METHODS_OVER_HTTP_ERROR_MESSAGE);
         }
     }
+
+    /**
+     * Sends the TelemetryMessage to IoT hub.
+     * @param telemetryMessage The user supplied telemetry message.
+     */
+    public void sendTelemetry(TelemetryMessage telemetryMessage)
+    {
+        sendTelemetry(telemetryMessage, null, null);
+    }
+
+    /**
+     * Sends the TelemetryMessage to IoT hub.
+     * @param telemetryMessage The user supplied telemetry message.
+     * @param callback the callback to be invoked when a response is received. Can be {@code null}.
+     * @param callbackContext a context to be passed to the callback. Can be {@code null} if no callback is provided.
+     */
+    public void sendTelemetry(TelemetryMessage telemetryMessage, IotHubEventCallback callback, Object callbackContext)
+    {
+        if (telemetryMessage == null)
+        {
+            throw new IllegalArgumentException("telemetryMessage property cannot be null");
+        }
+
+        if (telemetryMessage.Telemetry != null)
+        {
+            telemetryMessage.Telemetry.Convention = PayloadConvention;
+            telemetryMessage.setContentEncoding(PayloadConvention.getPayloadEncoder().getContentEncoding().name());
+            telemetryMessage.setContentTypeFinal(PayloadConvention.getPayloadSerializer().getContentType());
+        }
+
+        sendEventAsync(telemetryMessage, callback, callbackContext);
+    }
+
+    /**
+     * Set the global command callback handler.
+     * @param callback The callback to be used for commands.
+     * @param callbackContext An optional user context to be sent to the callback.
+     */
+    public void subscribeToCommands(DeviceMethodCallback callback, Object callbackContext)
+    {
+        // Subscribe to methods default handler internally and use the callback received internally to invoke the user supplied command callback.
+        // TODO Implement command handler
```

Comment:
I am actively working on this feature, just needed to get some eyes on this before it gets too long.